### DISCUSSION TITLE
GyroDriveBase to use_gyro problem

### DISCUSSION THREAD

**Discussion by Kermit647** [pybricks/support#1400] [General]
Hi,
I just updated our (spike) hubs to the newest not-beta version.
As expected the GyroDriveBase is not available anymore,
but it wont let us set use_gyro :

AttributeError: 'DriveBase' object has no attribute 'use_gyro'

Do we have to use the current beta ? Is gyro not included in the standart version ?

Thanks !

**Comment by laurensvalk** [comment]
If you run the following program, what do you see?

```python
from pybricks import version
print(version)
```

Can you share an example code that isn't working for you?

**Comment by Kermit647** [comment]
Wow, that was quick !  Code :

```python
from pybricks.hubs import PrimeHub
from pybricks.pupdevices import Motor, ColorSensor, UltrasonicSensor, ForceSensor
from pybricks.parameters import Button, Color, Direction, Port, Side, Stop
from pybricks.robotics import DriveBase
from pybricks.tools import wait, StopWatch
from pybricks import version
print(version)
left_motor = Motor(Port.E, Direction.COUNTERCLOCKWISE)
right_motor = Motor(Port.F)
drive_base = DriveBase(left_motor, right_motor, wheel_diameter=56, axle_track=160)
drive_base.use_gyro=True
```

Output :  

```console
('primehub', '3.3.0', 'v1.20.0-23-g6c633a8dd on 2023-11-24')
Traceback (most recent call last):
  File "Wro24.py", line 11, in <module>
AttributeError: 'DriveBase' object has no attribute 'use_gyro'
```

**Reply by laurensvalk** [reply]
> drive_base.use_gyro=True

This isn't right. The correct syntax should be:

```python
drive_base.use_gyro(True)
```

See the [docs](https://docs.pybricks.com/en/latest/robotics.html#driving-straight-and-turning-in-place) or [this post](https://github.com/orgs/pybricks/discussions/1399#discussioncomment-8253640) for examples on how you can use it.

> Wow, that was quick !

:smile:  :rocket:

**Reply by Kermit647** [reply]
Ok, thats really embarrassing. I will now just dig a hole for me and never come out again.
(At some point I even wondered why its a variable and no function as this would be bad design, but I never got the idea I copied it wrong somehow)

**Reply by laurensvalk** [reply]
No worries, it happens to the best of us! Good luck with WRO, and don't forget to tell other teams about Pybricks :smile:

**Comment by creativspelerr** [comment]
I am getting the same error regarding the gryo. If I use to check the attributes available to DriveBase it has things like turn and angle, but it does not have use_gyro. When I do from pybricks import version, I get an error "no module named 'pybricks.version'
Is there another way to check what version I have? It was just downloaded from the lego site.
DriveBase also has no attribute called "curve" if that helps identify it.

**Reply by dlech** [reply]
Since you said "It was just downloaded from the lego site.", I assume you are using EV3?

The LEGO Python for EV3 uses Pybricks 2.0 which doesn't have these newer additions.
